### PR TITLE
return false if remove failed

### PR DESCRIPTION
**Pull request by LukeOwncloud**
if OCFile is null --> cannot delete ==> return false.

### CODE REVIEW

File: src/com/owncloud/android/datamodel/FileDataStorageManager.java [hunk 1]
```diff
@@ -467,7 +467,7 @@ public void saveFolder(
     
 
     public boolean removeFile(OCFile file, boolean removeDBData, boolean removeLocalCopy) {
-        boolean success = true;
+        boolean success = false;
```
Comment (davivel) [inline src/com/owncloud/android/datamodel/FileDataStorageManager.java]: I'm sorry to say this change breaks the condition in line 497 for input parameters ```file != null && !file.isFolder()``` ```removeDBData == false```. 

I'd say the easiest way to achieve what you want is with adding an ```else``` block in line 509.

File: src/com/owncloud/android/datamodel/FileDataStorageManager.java [hunk 2]
```diff
@@ -512,7 +512,7 @@ public boolean removeFile(OCFile file, boolean removeDBData, boolean removeLocal
     
 
     public boolean removeFolder(OCFile folder, boolean removeDBData, boolean removeLocalContent) {
-        boolean success = true;
+        boolean success = false;
```
Comment (davivel) [inline src/com/owncloud/android/datamodel/FileDataStorageManager.java]: Similarly, this breaks the condition in line 520 for input parameters ```folder != null && folder.isFolder() ```, ```removeDBData == false ```. 

Once again, I'd go with an ```else``` block in line 523.

**Comment by davivel** [comment]
@LukeOwncloud , please, see my comments and make a new PR onto ```master ``` branch. Thanks a lot :)

**Comment by LukeOwncloud** [comment]
Very true!

**Pull request closed without merging**